### PR TITLE
[usm] Improve behavior of HTTPS slow-start

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

Add a best-effort fallback approach to monitor HTTPS traffic in connections that are older than system-probe. In other words, this allows system-probe (in certain cases) to monitor HTTPS traffic even when the TLS handshake is missed.

Thanks @guyarb for writing the test!

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/ebpf/c/https.h [hunk 1]
```diff
@@ -39,6 +39,18 @@ static __always_inline void https_finish(conn_tuple_t *t) {
...
+        // of data for TLS connections started *prior* to system-probe
+        // initialization.  here we simply store the pid_tgid along with its
+        // corresponding ssl_ctx pointer. in another probe (tcp_sendmsg), we
+        // query again this map and if there is a match we assume that the *sock
+        // object is the the TCP socket being used by this SSL connection.  The
+        // whole thing works based on the assumption that SSL_read/SSL_write is
+        // then followed by the execution of tcp_sendmsg within the same CPU
+        // context.  This is not necessarily true for all cases (such as when
+        // using the async SSL API) but seems to work on most-cases.
+        bpf_map_update_with_telemetry(ssl_ctx_by_pid_tgid, &pid_tgid, &ssl_ctx, BPF_ANY);
```
Comment (guyarb) [inline pkg/network/ebpf/c/https.h:53]: My main concern here is that we never delete entries from the map in that situation
I'd add a deletion in `init_ssl_sock_from_do_handshake` and change the name of the function to be more accurate with the current change

**Review by guyarb — DISMISSED** [review]

**Comment by guyarb** [comment]
Rebased to main and added a commit with a fix

File: pkg/network/tracer/tracer_test.go [hunk 2]
```diff
@@ -1950,6 +1951,95 @@ func TestOpenSSLVersions(t *testing.T) {
...
+				// reqIndex is 0 based, while the number is requests[reqIndex] is 1 based.
+				t.Logf("request %d was not found (req %v)", reqIndex+1, requests[reqIndex])
+				return false
+			}
+		}
+
+		return true
+	}, 3*time.Second, time.Second, "connection not found")
+
+	for reqIndex, exists := range missedRequestsExist {
```
Comment (amitslavin) [inline pkg/network/tracer/tracer_test.go:2038]: The way you use the the var exists is confusing, in the start you iterate over requestExist and used exists, now you iterate over missedRequestExist and again the var is named exists? I think you should use better naming for both of the loops in order to make it more clear

Comment (guyarb) [reply pkg/network/tracer/tracer_test.go:2038]: correct, that's a copy-paste bug
modified

**Review by amitslavin — APPROVED** [review]

**Review by hmahmood — APPROVED** [review]

**Pull request merged**